accountsdb-plugin-manager/src/accountsdb_plugin_service.rs

```diff
@@ -74,13 +74,27 @@ impl AccountsDbPluginService {
         for accountsdb_plugin_config_file in accountsdb_plugin_config_files {
             Self::load_plugin(&mut plugin_manager, accountsdb_plugin_config_file)?;
         }
+        let to_notify_account_data = plugin_manager.to_notify_account_data();
 
         let plugin_manager = Arc::new(RwLock::new(plugin_manager));
-        let accounts_update_notifier = Arc::new(RwLock::new(AccountsUpdateNotifierImpl::new(
-            plugin_manager.clone(),
-        )));
-        let slot_status_observer =
-            SlotStatusObserver::new(confirmed_bank_receiver, accounts_update_notifier.clone());
+
+        let accounts_update_notifier: Option<AccountsUpdateNotifier> = if to_notify_account_data {
+            let accounts_update_notifier = AccountsUpdateNotifierImpl::new(plugin_manager.clone());
+            Some(Arc::new(RwLock::new(accounts_update_notifier)))
+        } else {
+            None
+        };
+
+        let slot_status_observer = if to_notify_account_data {
```

Comment:
Should this have a different if case?
I'm guessing you might be envisioning that ultimately plugins can opt-in to account-data, slot-data, and/or tx-data separately. Is that right?

Reply:
When account or tx data is requested -- it implicitly requires the slot status data. So in the future it will look like let slot_status_observer = if to_notify_account_data || to_notify_transaction_data  {

Reply:
Ah okay, got it, thanks.